rust/update-conan-recipe-and-create-pull-request/src/decode_github_content.rs

```diff
@@ -1,17 +1,29 @@
-use base64::Engine;
-use crate::add_version_to_conandata_yml::AddVersionToConandataYmlArgument;
+use {crate::Result, base64::DecodeError};
 
+#[deprecated]
 pub struct DecodeGithubContentArgument<'a> {
     pub content: &'a String,
 }
 
-pub fn decode_github_content(argument: &DecodeGithubContentArgument<'_>) -> Option<String> {
-    let DecodeGithubContentArgument {
-        content
-    } = argument;
```

Comment:
You can match in fn sig - any `let .. = ..` and `..: Type` is always pattern matching
[`let expr`](https://docs.rs/syn/latest/syn/struct.ExprLet.html) and [`fn arg`](https://docs.rs/syn/latest/syn/struct.PatType.html)